Extended console application

fhaag opened the pull request:
Along with my changes for adding the [placeholder feature](https://github.com/Knagis/CommonMark.NET/pull/93), I have slightly extended my local copy of the console application to simplify checking how the additional features and the non-slim ("fat") HTML formatter work. Feel free to integrate these, too, if you deem these changes useful.

Knagis commented on CommonMark.Console/Program.cs:
```diff
@@ -61,6 +61,19 @@ static int Main(string[] args)
                     {
                         settings.OutputFormat = OutputFormat.SyntaxTree;
                     }
+                    else if (string.Equals(args[i], "--fat", StringComparison.OrdinalIgnoreCase))
```
perhaps it would be simpler to automatically imply --fat when --extended is specified? Since the only reason why someone would want to use --fat is to use the placeholders that require the fat formatter.

fhaag commented:
That depends a bit on the intended purpose of the console application. While modifying the library, I used it as a convenient entry point to reach breakpoints anywhere in the CommonMark.NET library. Hence, I had an incentive to explicitly control which features and modules to use, even if the output didn't change.

If the console application as included in the releases is just meant for performing the transformation, I can modify the code so as to automatically use the non-slim converter when `--extended` is specified.

At least in the latter case, I would, however, like to offer another command line argument, for specifying some kind of key/value pairs (maybe as a colon-separated list or similar). If present, these could be applied by the placeholder feature. Like this, the placeholder feature could at least be used in a limited extent (for a 1:1 substitution) from the console application.

fhaag commented:
I have removed the explicit `--fat` argument and integrated its functionality into the `--extended` argument.

Furthermore, I have added the described placeholder testing feature by means of a `--subst` argument. Try

    CommonMark.Console.exe --extended --subst known:replaced test.md

for a `test.md` file with contents like

    There are [known] and [unknown] placeholders.

Knagis commented on CommonMark.Console/Program.cs:
```diff
@@ -61,6 +66,32 @@ static int Main(string[] args)
+                        {
+                            var tokens = args[i + 1].Split(':');
+
+                            if (tokens.Length % 2 == 0)
```
an error should be shown if this does not pass

fhaag replied:
Done, now returning `1` after invoking `PrintUsage` again.

Knagis commented on CommonMark.Console/Program.cs:
```diff
@@ -140,6 +171,34 @@ static int Main(string[] args)
+                        settings.OutputDelegate = (block, writer, cmSettings) =>
+                        {
+                            var formatter = new Formatters.HtmlFormatter(writer, cmSettings);
+                            formatter.PlaceholderResolver = placeholderText =>
```
wrap just this assignment in `if(placeholderSubstitutionTable==null)` so that if other flags are added it would be easier to extend the code.

fhaag replied:
I noticed if a benchmark is run, this distinction would be within the benchmarked code, but as that particular instruction should be hit only once, I was probably overoptimizing. Changed.

Knagis commented:
I merged the PR however if you have time the two comments should be addressed.
